lib/active_model/validations/store_model_validator.rb

```diff
@@ -20,19 +20,15 @@ def validate_each(record, attribute, value)
           return
         end
 
-        case record.type_for_attribute(attribute).type
+        record_type = record.type_for_attribute(attribute).type
+        strategy = StoreModel::CombineErrorsStrategies.configure(options, record_type)
+        case record_type
         when :json
           strategy.call(attribute, record.errors, value.errors) if value.invalid?
         when :array
-          record.errors.add(attribute, :invalid) if value.select(&:invalid?).present?
+          strategy.call(attribute, record.errors, value) if value.select(&:invalid?).present?
```

Comment:
The difference int he call api is kind of annoying. I'm trying to avoid having to have to duplicate all the config, on Lambda's that you'd have to configure to support both singular and array types. For any given model/attribute only one is in play. However, the global config versions need to support both somehow. 

An option would be to have the :json strategy call with the value and not value.errors. That would be a breaking change. Alternatively we just duplicate all the configuration options in parallel.